eng/pipelines/npm-tasks.yml

```diff
@@ -51,33 +51,47 @@ extends:
   parameters:
     stages:
       - stage:
-        displayName: NPM Task Execute
+        displayName: 'NPM Task Execute'
+        variables:
+          - template: /eng/pipelines/templates/variables/image.yml
 
         jobs:
-        - deployment: 'NPM_Admin'
-          displayName: NPM package management
-          environment: npm
-
+        - job: NPM_Admin
+          displayName: 'NPM package management'
           pool:
-            name: azsdk-pool-mms-win-2022-general
-            image: azsdk-pool-mms-win-2022-1espt
+            name: $(WINDOWSPOOL)
+            image: $(WINDOWSVMIMAGE)
             os: windows
 
+          steps:
+            - checkout: self
+            - download: current
+            - task: PowerShell@2
+              displayName: 'Run Task'
+              inputs:
+                targetType: filePath
+                filePath: "$(System.DefaultWorkingDirectory)/eng/scripts/npm-admin-tasks.ps1"
+                arguments: >
+                  -taskType ${{parameters.TaskType}}
+                  -packageName ${{parameters.PackageName}}
+                  -pkgVersion ${{parameters.PkgVersion}}
+                  -tagName ${{parameters.TagName}}
+                  -npmToken "$(azure-sdk-npm-token)"
+                  -reason "${{parameters.Reason}}"
+                pwsh: true
+
+        # This is a workaround to not run admin job in deployment job but still have the approval gate
+        - deployment: NPMSuccess
+          displayName: NPM package management success
+          environment: npm
+          dependsOn: NPM_Admin
+          condition: succeeded()
+
           strategy:
```

Comment:
We should use a similar approach as https://github.com/Azure/azure-sdk-for-ios/blob/main/eng/pipelines/templates/stages/archetype-ios-release.yml#L25

Reply:
If we even want to have this require an extra approval. I'm not sure we even need a deployment job for this.

Reply:
Seems like we could use https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/manual-validation-v1?view=azure-pipelines for approval, is it ok if the permission to approve is whoever has permission to queue the pipeline?

Reply:
We could consider a manual validation but we haven't really used it and don't know what issues it may or may not have. I would like to stick with environments the deployments so we have exactly one place to configure the approval settings.

Reply:
okok